src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -2787,3 +2789,115 @@ def find_q(m, m4_fac, D):
                         yield Et
                     except ValueError:
                         pass
+
+def EllipticCurve_with_prime_order(N):
+    r"""
+    Given a prime number ``N``, find another prime number `p` and construct an
+    elliptic curve `E` defined over `\mathbb F_p` such that
+    `\#E(\mathbb F_p) = N`.
+
+    INPUT:
+
+    - ``N`` -- integer; the order for which we seek an elliptic curve. Must be a
+      prime number.
+
+    OUTPUT: an elliptic curve `E/\mathbb F_p` of order ``N``
+
+    EXAMPLES::
+
+        sage: N = next_prime(int(b'sagemath'.hex(), 16))
+        sage: E = EllipticCurve_with_prime_order(N)
+        sage: E
+        Elliptic Curve defined by y^2 = x^3 + 4757897140353078952*x +
+        1841350074072114366 over Finite Field of size 8314040074357871443
+        sage: E.order() == N
+        True
+
+    The execution time largely depends on the input::
+
+        sage: N = 125577861263605878504082476745517446213
+        sage: E = EllipticCurve_with_prime_order(N) # Takes ~1 second.
+        sage: E.order() == N
+        True
+
+    TESTS::
+
+        sage: N = next_prime(123456789)
+        sage: E = EllipticCurve_with_prime_order(N)
+        sage: E.order() == N
+        True
+
+        sage: for N in prime_range(3, 100):
+        ....:     E = EllipticCurve_with_prime_order(N)
+        ....:     assert E.order() == N
+
+        sage: N = 123456789
+        sage: E = EllipticCurve_with_prime_order(N)
+        Traceback (most recent call last):
+        ...
+        ValueError: input order is not prime
+
+        sage: E = EllipticCurve_with_prime_order(0)
+        Traceback (most recent call last):
+        ...
+        ValueError: input order is not prime
+
+    .. NOTE::
```

Comment:
```suggestion
    .. WARNING::
```

Reply:
I'm not sure what's the fundamental difference here? It doesn't feel like a warning, same as `factor()` can take a very long time?

Reply:
The way it's worded ("function may run forever") makes it sound like the algorithm isn't proven to always terminate.

If it does terminate but possibly only after a very long time, then I think `NOTE` is okay. In that case word the note to say "long time" or something like that. If there are certain inputs that make it more likely to take a long time (for example, very large numbers) then it would be nice if the note mentioned which kinds of cases cause it to take a long time.

Reply:
I was not so fine either with it and the fact that you point it out confirms it. What about fddd496e474a73745a9267e9f745f67dd6b24612 ?

Reply:
This is great, just adding a suggestion for formatting.